--- apg.c
+++ apg.c
@@ -571,7 +571,7 @@
  char * seq;
  UINT32 prom[2] = { 0L, 0L };
  UINT32 sdres = 0L;
- printf ("\nPlease enter some random data (only first %d are significant)\n", sizeof(prom));
+ printf ("\nPlease enter some random data (only first %d are significant)\n", (int)sizeof(prom));
  seq = (char *)getpass("(eg. your old password):>");
  if (strlen(seq) < sizeof(prom))
   (void)memcpy((void *)&prom[0], (void *)seq, (int)strlen(seq));
--- Makefile
+++ Makefile
@@ -1,19 +1,9 @@
 ##################################################################
-# You can modify CC variable if you have compiler other than GCC
-# But the code was designed and tested with GCC
-CC = gcc
-
-##################################################################
-# Compilation flags
-# You should comment the line below for AIX+native cc
-FLAGS = -Wall
-
-##################################################################
 # Libraries
 #
 # You should comment the line below ('LIBS= -lcrypt')for QNX RTP
 # 6.1.0, OpenBSD 2.8 and above, WIN32 (+MinGW)
-LIBS = -lcrypt
+#LIBS = -lcrypt
 LIBM = -lm
 # Use lines below for cygwin
 # LIBS = 
@@ -39,7 +29,7 @@
 ##################################################################
 # Directories
 # Install dirs
-INSTALL_PREFIX = /usr/local
+APG_PREFIX = /usr/local
 APG_BIN_DIR = /bin
 APG_MAN_DIR = /man/man1
 APGD_BIN_DIR = /sbin
@@ -74,9 +64,6 @@
 #CS_LIBS = -lsocket
 
 # ====== YOU DO NOT NEED TO MODIFY ANYTHING BELOW THIS LINE ======
-# Find group ID for user root
-FIND_GROUP = `grep '^root:' /etc/passwd | awk -F: '{ print $$4 }'`
-
 PROGNAME = apg
 CS_PROGNAME = apgd
 BFM_PROGNAME = apgbfm
@@ -96,52 +83,54 @@
 cygwin: standalone
 
 apg:
-	${CC} ${FLAGS} -D${CRYPTED_PASS} -D${USE_SHA} -o ${PROGNAME} ${SOURCES} ${LIBS} ${LIBM}
+	${CC} ${CPPFLAGS} ${CFLAGS} -D${CRYPTED_PASS} -D${USE_SHA} -o ${PROGNAME} ${SOURCES} ${LDFLAGS} ${LIBS} ${LIBM}
 
 apgd:
-	${CC} ${FLAGS} -DCLISERV -D${USE_SHA} -o ${CS_PROGNAME} ${SOURCES} ${CS_LIBS} ${LIBM}
+	${CC} ${CPPFLAGS} ${CFLAGS} -DCLISERV -D${USE_SHA} -o ${CS_PROGNAME} ${SOURCES} ${LDFLAGS} ${CS_LIBS} ${LIBM}
 
 apgbfm:
-	${CC} ${FLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${LIBM}
+	${CC} ${CPPFLAGS} ${CFLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${LDFLAGS} ${LIBM}
 
 strip:
 	strip ${PROGNAME}
 	strip ${CS_PROGNAME}
 	strip ${BFM_PROGNAME}
 
+.PHONY: install
 install:
 	if test -x ./apg; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apg ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apg.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
+./install-sh -c -m 0755 ./apg ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./install-sh -c -m 0444 ./doc/man/apg.1 ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
 fi
 	if test -x ./apgd; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgd ${INSTALL_PREFIX}${APGD_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apgd.8 ${INSTALL_PREFIX}${APGD_MAN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APGD_BIN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APGD_MAN_DIR}; \
+./install-sh -c -m 0755 ./apgd ${DESTDIR}${APG_PREFIX}${APGD_BIN_DIR}; \
+./install-sh -c -m 0444 ./doc/man/apgd.8 ${DESTDIR}${APG_PREFIX}${APGD_MAN_DIR}; \
 fi
 	if test -x ./apgbfm; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 -o root -g ${FIND_GROUP} ./apgbfm ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
+./install-sh -c -m 0755 ./apgbfm ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
 fi
 
 install-cygwin:
 	if test -x ./apg.exe; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 ./apg.exe ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apg.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
+./install-sh -c -m 0755 ./apg.exe ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./install-sh -c -m 0444 ./doc/man/apg.1 ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
 fi
 	if test -x ./apgbfm.exe; then \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./mkinstalldirs ${INSTALL_PREFIX}${APG_MAN_DIR}; \
-./install-sh -c -m 0755 ./apgbfm.exe ${INSTALL_PREFIX}${APG_BIN_DIR}; \
-./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${INSTALL_PREFIX}${APG_MAN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./mkinstalldirs ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
+./install-sh -c -m 0755 ./apgbfm.exe ${DESTDIR}${APG_PREFIX}${APG_BIN_DIR}; \
+./install-sh -c -m 0444 ./doc/man/apgbfm.1 ${DESTDIR}${APG_PREFIX}${APG_MAN_DIR}; \
 fi
 
+.PHONY: clean
 clean:
 	rm -f ${CS_PROGNAME} ${PROGNAME} ${BFM_PROGNAME} ${OBJECTS} *core*
