--- tests/runtests.c.orig	Fri Feb 18 17:41:08 2022
+++ tests/runtests.c	Wed Feb 23 11:03:49 2022
@@ -60,6 +60,16 @@
 #  include <stdbool.h>
 #endif
 
+#if defined(__sun__)
+#  include <sunmath.h>
+#  if ! defined(INFINITY)
+#    define INFINITY (__builtin_huge_val())
+#  endif
+#  if ! defined(NAN)
+#    define NAN (__builtin_nan(""))
+#  endif
+#endif
+
 #include "expat.h"
 #include "chardata.h"
 #include "structdata.h"
@@ -6636,7 +6646,8 @@
   XML_Char **elemstr = (XML_Char **)userData;
   if (xcstrcmp(elemstr[0], name) != 0) {
     char buffer[1024];
-    sprintf(buffer, "unexpected end string: '%" XML_FMT_STR "'", name);
+    snprintf(buffer, sizeof(buffer), "unexpected end string: '%" XML_FMT_STR "'", name);
+    buffer[sizeof(buffer)-1] = '\0';  /* ensure NULL termination */
     fail(buffer);
   }
   triplet_end_flag = XML_TRUE;
