--- p11-kit/lists.c
+++ p11-kit/lists.c
@@ -43,12 +43,20 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <stdint.h>
+#include <limits.h>
 
 #include "message.h"
 #include "p11-kit.h"
 #include "tool.h"
 #include "uri.h"
 
+#ifndef SIZE_MAX
+# ifdef SIZE_T_MAX
+#  define SIZE_MAX SIZE_T_MAX
+# endif
+#endif
+
 int p11_kit_list_modules (int argc,
                           char *argv[]);
 
